sdcm/tester.py

```diff
@@ -883,70 +884,89 @@ def setUp(self):  # pylint: disable=too-many-branches,too-many-statements
-                                                password=self.params.get('authenticator_password'))
-                        # TODO: update proper loader and monitor in multi-tenant case.
-                        #       Now it updates only the first ones all the time.
-                        self.loaders.added_password_suffix = True
-                        # TODO: Replace with strong password generation
-                        self.monitors.added_password_suffix = True
-            else:
-                self.set_system_auth_rf(db_cluster=db_cluster)
-
+        def _create_loaders():
             if self.loaders and not self.loaders_multitenant:
                 self.loaders_multitenant = [self.loaders]
             for loaders in self.loaders_multitenant:
                 if loaders:
                     loaders.wait_for_init()
 
-        # cs_db_cluster is created in case MIXED_CLUSTER. For example, gemini test
-        if self.cs_db_cluster:
-            init_value = self.params.get("use_mgmt")
-            self.params["use_mgmt"] = False
-            self.init_nodes(db_cluster=self.cs_db_cluster)
-            self.params["use_mgmt"] = init_value
+        def _create_gemini_cluster():
+            # cs_db_cluster is created in case MIXED_CLUSTER. For example, gemini test
+            if self.cs_db_cluster:
+                self.init_nodes(db_cluster=self.cs_db_cluster)
+
+        def _create_monitors():
+            if self.monitors and not self.monitors_multitenant:
+                self.monitors_multitenant = [self.monitors]
+
+            if self.monitors_multitenant:
+                monitors_init_in_parallel = ParallelObject(
+                    timeout=3600,
+                    objects=[[monitor] for monitor in self.monitors_multitenant],
+                    num_workers=len(self.monitors_multitenant))
+                monitors_init_in_parallel.run(
+                    func=(lambda m: m.wait_for_init()),
+                    unpack_objects=True, ignore_exceptions=False)
+
+        def _validate_raft():
+            for db_cluster in self.db_clusters_multitenant:
+                if db_cluster:
+                    db_cluster.validate_seeds_on_all_nodes()
+                    validate_raft_on_nodes(nodes=db_cluster.nodes)
+                    db_cluster.start_kms_key_rotation_thread()
+
+        with ThreadPoolExecutor(max_workers=4) as executor:
+            futures = [
+                executor.submit(_create_db_clusters),
+                executor.submit(_create_loaders),
+                executor.submit(_create_monitors),
+                executor.submit(_create_gemini_cluster)
+            ]
+
+            for future in as_completed(futures):
+                future.result()
+
+        threading.Thread(target=_validate_raft).start()
+        threading.Thread(target=self.argus_collect_packages).start()
+        threading.Thread(target=self.argus_get_scylla_version).start()
```

Comment:
Error verification?

Reply:
*Let's check if _validate_raft is needed to be wait for,
* lets try moving to using  `ThreadPoolExecutor`

Reply:
ok, I decided for now to move this under ThreadPoolExecutor as running it in background thread and not waiting or result requires analysis. This will be done in future work for speeding up setup.